docs/performance/remote-filesystem/amazon-s3.md

```diff
@@ -0,0 +1,396 @@
+---
+title: Amazon S3
+level: advanced
+---
+Amazon S3
+=========
+
+Configuring Bolt to use Amazon Simple Storage Service (Amazon S3) requires the
+addition of Flysystem and caching libraries, as well as custom service provider
+code to your project.
+
+1. Set-up, configure, and confirm operational your Amazon S3 bucket
+1. Update `composer.json` in your project root
+1. Creating service providers
+1. Bootstrap your providers
+
+Prerequisites
+-------------
+
+<p class="warning"><strong>Warning:</strong> This documentation, and examples,
+assume you have a fully configured and working Amazon S3 bucket.
+</p>
+
+
+
+### Required Libraries
+----------------------
+
+You need to add the following libraries to your project:
+
+  * `league/flysystem-aws-s3-v3:^1.0`
+  * `league/flysystem-cached-adapter:^1.0`
+  * `league/flysystem-memory:^1.0`
+
+This can simply be achieved by running:
+
+```bash
+$ composer require league/flysystem-aws-s3-v3:^1.0 league/flysystem-cached-adapter:^1.0 league/flysystem-memory:^1.0
+```
+
+
+### Caching Libraries
+----------------------
+
+While other cache targets are optional, right now this example will be base on Redis.
+
+To use Redis, you will need a configured and running Redis instance, and need
+to install `predis/predis:^1.1`, which can be achieved by running:
+
+
+```bash
+$ composer require predis/predis:^1.1
+```
+
+
+Autoloading
+------------
+
+Autoloading of your service providers can be set-up in the same way as
+[Extension Bundles][bundles], by adding your base namespace and source
+directory ot the `psr-4` section of your project root `composer.json` file,
+e.g.
+
+```json
+    "autoload": {
+        "psr-4": {
+            "MySiteApp\\": "src"
+        }
+    }
+```
+
+
+<p class="note"><strong>Note:</strong> If you change anything in the
+<code>psr-4</code> section of your <code>composer.json</code> file, remember to
+re-run <code>composer dump-autoload</code> to update the cached autoloading
+files.
+</p>
```

Comment:
I'd say it should stay in. People are very good at just skipping over stuff they don't need, so there's no real harm in keeping it. Alternatively, for a bunch of people this is not trivial. :-)